### PR TITLE
Endpoints test xunit

### PR DESCRIPTION
**Pull request by SurferJeffAtGoogle**

### CODE REVIEW

File: endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using Xunit;
+using Newtonsoft.Json;
+using System.IO;
+
+namespace IO.SwaggerTest
+{
+    public class UnitTest1
```
Comment (simonz130) [inline endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Can you rename it to something more meaningful? File name as well.

File: endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        class Payload
+        {
+            public string Message { get; set; }
+        }
+
+        [Fact]
+        public async void TestEcho()
+        {
+            var request = HttpWebRequest.Create("http://localhost:7412/echo");
```
Comment (simonz130) [inline endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Can you read this from configuration?

Comment (simonz130) [inline endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: We should use HttpClient to make http requests in .NET Core code. HttpWebRequest is old, outdated API.

File: endpoints/getting-started/src/runTests.ps1 [hunk 3]
```diff
@@ -12,18 +12,18 @@
...
-$url = "http://localhost:7412"
-$job = Run-Kestrel($url)
-Start-Sleep -Seconds 5
+Import-Module -DisableNameChecking ..\..\..\BuildTools.psm1
+Set-PSDebug -Trace 1
 try {
-	.\Test.ps1 "$url/echo"
+	Push-Location
+	Set-Location IO.Swagger
+	$url = "http://localhost:7412"
```
Comment (simonz130) [inline endpoints/getting-started/src/runTests.ps1]: Can we have this and the http client request in tests read this value from the same location?

File: BuildTools.psm1 [hunk 4]
```diff
@@ -770,31 +789,26 @@ function Move-TestResults($OutDir) {
...
+        $LASTEXITCODE = 0      
+        foreach ($suite in $x.testsuites.testsuite) {
+            $LASTEXITCODE += [int] $suite.failures 
         }
-        $sleepSeconds *= 2
+    } else {
+        $casperOut = casperjs $TestJs $Url
+    }
+    if ($LASTEXITCODE -eq 0) {
+        $casperOut | Write-Host
```
Comment (simonz130) [inline BuildTools.psm1]: Is Casper output tagged somehow so it's easy to recognize that it's casper output?

File: endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            request.Method = "POST";
+            request.ContentType = "application/json";
+            using (var streamWriter = new StreamWriter(request.GetRequestStream()))
+            {
+                streamWriter.Write(JsonConvert.SerializeObject(
+                    new Payload { Message = "in a bottle" }));
```
Comment (simonz130) [inline endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: nit: create a const string at the start of the test, then use it here and in assert later on.

Comment (SurferJeffAtGoogle) [reply endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Done.

File: BuildTools.psm1 (hunk 4 above)
Comment (SurferJeffAtGoogle) [reply BuildTools.psm1]: Casper also creates a TestResults.xml, which gets rendered in TestFusion and Sponge.

File: endpoints/getting-started/src/runTests.ps1 (hunk 3 above)
Comment (SurferJeffAtGoogle) [reply endpoints/getting-started/src/runTests.ps1]: Done.

File: endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs (hunk 2 above)
Comment (SurferJeffAtGoogle) [reply endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Done.

Comment (SurferJeffAtGoogle) [reply endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Done.

File: endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs (hunk 1 above)
Comment (SurferJeffAtGoogle) [reply endpoints/getting-started/src/IO.SwaggerTest/UnitTest1.cs]: Done.

**Comment by SurferJeffAtGoogle** [comment]
> Do we really need casper and kestrel to test what we test? Can't we just leverage the fact it's ASP.NET Core OWIN and create integration test with the same effect?
> https://docs.microsoft.com/en-us/aspnet/core/testing/integration-testing?view=aspnetcore-2.1

For this test, casper is not needed and not used.

For other tests, we need to do much more than just fetch pages and check http status codes.  We need to click on links in pages.  We need to run the javascript and observe changes to page elements.  ASP.NET's test server does not appear to be able to do that.

File: BuildTools.psm1 (hunk 4 above)
Comment (simonz130) [reply BuildTools.psm1]: I probably didn't structure the question well. 
I meant if there is any way to distinguish between the traces that are created by casper and the traces that are created by other parts (like dotnet test, ASP.NET service traces, etc)? Assuming that all the traces will be captured in testResults.xml

Also, not relevant for this code review, but in general, in my opinion, it's "dotnet test" that should produce TestResults.xml via custom logger implementation we should supply it. Casper traces should be captured by the test harness (dotnet test) via custom logger.

**Pull request merged**